editoast/src/generated_data/error/mod.rs

```diff
@@ -246,30 +268,117 @@ fn get_insert_errors_query(obj_type: ObjectType) -> &'static str {
     }
 }
 
-/// Insert a heterogeneous list of infra errors in DB with a minimum number of queries
-async fn insert_errors(
-    conn: &mut PgConnection,
-    infra_id: i64,
-    errors: Vec<InfraError>,
-) -> Result<()> {
-    let mut errors_by_type: HashMap<_, Vec<_>> = Default::default();
+/// This function dispatch errors to the right layer
+fn dispatch_errors_to_layers(
+    errors: Vec<ErrorWithHash>,
+) -> HashMap<ObjectType, Vec<ErrorWithHash>> {
+    let mut errors_by_layer: HashMap<_, Vec<_>> = Default::default();
     for error in errors {
-        errors_by_type
-            .entry(error.get_type())
+        errors_by_layer
+            .entry(error.error.get_type())
             .or_default()
-            .push(to_value(error).unwrap());
+            .push(error);
     }
+    errors_by_layer
+}
+
+/// Retrieve the current errors hash for a given infra
+async fn retrieve_current_errors_hash(
+    conn: &mut PgConnection,
+    infra_id: i64,
+) -> Result<Vec<String>> {
```

Comment:
Would it make sense to return a more specific type than `String`? Like a specific `Hash` type, or at least a `NonBlankString`?

Reply:
We could use `[u8; 20]` instead of `String` but this means we expect a valid sha1 value in DB.
I can define a `type ErrorHash = String` for now so it will be easier to change it later.